src/language/HTMLSimpleDOM.js

```diff
@@ -75,19 +79,49 @@ define(function (require, exports, module) {
         section : { p: true },
         table   : { p: true },
         ul      : { p: true },
-        rt      : { rp: true, rt: true },
-        rp      : { rp: true, rt: true },
+        rb      : { rb: true, rt: true, rtc: true, rp: true },
+        rp      : { rb: true, rt: true, rp: true },
+        rt      : { rb: true, rt: true, rp: true },
+        rtc     : { rb: true, rt: true, rtc: true, rp: true },
         optgroup: { optgroup: true, option: true },
         option  : { option: true },
-        tbody   : { thead: true, tbody: true, tfoot: true },
-        tfoot   : { tbody: true },
-        tr      : { tr: true, th: true, td: true },
+        tbody   : { caption: true, colgroup: true, thead: true, tbody: true, tfoot: true, },
+        tfoot   : { caption: true, colgroup: true, thead: true, tbody: true },
+        thead   : { caption: true, colgroup: true },
+        tr      : { tr: true, th: true, td: true, caption: true },
         th      : { th: true, td: true },
-        td      : { thead: true, th: true, td: true },
-        body    : { head: true, link: true, script: true }
+        td      : { th: true, td: true },
+        body    : { head: true }
     };
 
     /**
+     * A list of elements which are automatically closed when their parent is closed:
+     * http://www.w3.org/html/wg/drafts/html/master/syntax.html#optional-tags
+     */
+    var optionalClose = {
+      html: true,
+      body: true,
+      li: true,
+      dd: true,
```

Comment:
Add `dt` here too, it's not completely correct behavior but without it it flickers around syntax error every other keypress. See the https://github.com/adobe/brackets/issues/7257 case for an example